src/text_shaper/directwrite_shaper.cpp

```diff
@@ -140,70 +140,84 @@ optional<font_key> directwrite_shaper::load_font(font_description const& _descri
     std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> wStringConverter;
     std::wstring familyName = wStringConverter.from_bytes(_description.familyName);
 
-    for (UINT32 i = 0, e = fontCollection->GetFontFamilyCount(); i < e; ++i)
-    {
-        IDWriteFontFamily* family{};
-        fontCollection->GetFontFamily(i, &family);
+    UINT32 familyIndex;
+    BOOL familyExists = FALSE;
+    fontCollection->FindFamilyName(familyName.data(), &familyIndex, &familyExists);
+
+    if (!familyExists) {
+        // Fallback to Consolas
+        const wchar_t* consolas = L"Consolas";
+        fontCollection->FindFamilyName(consolas, &familyIndex, &familyExists);
+    }
 
-        IDWriteLocalizedStrings* names;
-        family->GetFamilyNames(&names);
+    Microsoft::WRL::ComPtr<IDWriteFontFamily> fontFamily;
+    fontCollection->GetFontFamily(familyIndex, &fontFamily);
 
-        BOOL exists = FALSE;
-        unsigned index{};
-        names->FindLocaleName(d->userLocale.c_str(), &index, &exists);
+    Microsoft::WRL::ComPtr<IDWriteLocalizedStrings> familyNames;
+    fontFamily->GetFamilyNames(&familyNames);
 
-        wchar_t name[64];
-        if (exists)
-            names->GetString(index, name, _countof(name));
-        else
-            // TODO: there was a different way of fallback that I have a tab open on
-            // one of my smartphones. Please find me.
-            names->GetString(i, name, sizeof(name));
+    BOOL localeExists = FALSE;
+    unsigned index{};
+    familyNames->FindLocaleName(d->userLocale.c_str(), &index, &localeExists);
 
-        if (familyName != name)
-            continue;
+    if (!localeExists) {
+        // Fallback to en-US
+        const wchar_t* localeName = L"en-US";
```

Comment:
This is quite important because in CJK environment the locale will be something like `zh-Hans-HK` and most programming fonts do not have that locale available.